From c7f4e4485a35b585742c4252816a494eaec1e87c Mon Sep 17 00:00:00 2001
From: Bharat Potnuri <bharat@chelsio.com>
Date: Wed, 31 Jan 2018 21:07:51 +0530
Subject: [PATCH rdma-core] iwpmd: fix double mutex unlock

pthread_mutex_unlock() is used twice and this patch fixes it.

Fixes: be3fbf85b ("iwpmd: use ccan list.h for pending_messages")
Signed-off-by: Potnuri Bharat Teja <bharat@chelsio.com>
Reviewed-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
---
 iwpmd/iwarp_pm_helper.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/iwpmd/iwarp_pm_helper.c b/iwpmd/iwarp_pm_helper.c
index bebc151f..b8c23018 100644
--- a/iwpmd/iwarp_pm_helper.c
+++ b/iwpmd/iwarp_pm_helper.c
@@ -578,7 +578,6 @@ int add_iwpm_pending_msg(iwpm_send_msg *send_msg)
 
 	pthread_mutex_lock(&pending_msg_mutex);
 	list_add(&pending_messages, &pending_msg->entry);
- 	pthread_mutex_unlock(&pending_msg_mutex);
 	pthread_mutex_unlock(&pending_msg_mutex);
 	/* signal the thread that a new message has been posted */
 	pthread_cond_signal(&cond_pending_msg);
-- 
2.16.1

